### ISSUE TITLE
ignoring some ports not working (no errors too)

### ISSUE THREAD

**Issue by harshith-red** [rofl0r/proxychains-ng#396]
hello, im having an issue with ignoring a port
i put this in my conf file:

`localnet 0.0.0.0:80/0.0.0.0`

and when i run:

`proxychains4 curl ifconfig.me`

i get my squid server's ip address and not mine, which means that the port is being proxied. 
output:
![proxy](https://user-images.githubusercontent.com/70360354/130899476-3340181d-464e-4892-8edd-c45a44ee8ec7.jpg)
the file:
![image](https://user-images.githubusercontent.com/70360354/130899641-b4554c88-0917-4fb9-9f0b-c976380d6cb1.png)

**Comment by NingYuanLin** [comment]
I also find same problem as you in version 4.15, but it don't appear in version 4.12.

**Comment by rofl0r** [comment]
> I also find same problem as you in version 4.15, but it don't appear in version 4.12.

could you try to use `git bisect` to find the commit that broke it ?

**Comment by NingYuanLin** [comment]
> > I also find same problem as you in version 4.15, but it don't appear in version 4.12.
> 
> could you try to use `git bisect` to find the commit that broke it ?

I just tried it, and I found that the bug appeared in this commit:
```
(ning_base) ning@ubuntu:/usr/local/src/temp/proxychains-ng$ sudo git bisect good
50c079d4b44d6d96064fd0985306e165c2ed8285 is the first bad commit
commit 50c079d4b44d6d96064fd0985306e165c2ed8285
Author: rofl0r <retnyg@gmx.net>
Date:   Thu Feb 28 13:08:02 2019 +0000

    don't try to connect to null-routed ips

:040000 040000 4dd89242b410239e0c0ab2dfa1f8197c27840650 e6bd84663116812a7a5e9cbc6014640c1c0558f7 M      src
```
At the same time, I must point out that there is a problem with the test method of the original issue, because `localnet 0.0.0.0:80/0.0.0.0` is only valid for ip instead of domain name.  
I tested it according to the following method：  
1. `python -m http.server 2345`
2. `proxychains4 wget 0.0.0.0:2345`

**Comment by rofl0r** [comment]
>    don't try to connect to null-routed ips

thanks, good find.

>  I tested it according to the following method

what was the config for localnet for your test ?

**Comment by NingYuanLin** [comment]
> 

localnet 0.0.0.0:2345/0.0.0.0

**Comment by NingYuanLin** [comment]
> > don't try to connect to null-routed ips
> 
> thanks, good find.
> 
> > I tested it according to the following method
> 
> what was the config for localnet for your test ?

localnet 0.0.0.0:2345/0.0.0.0

**Comment by rofl0r** [comment]
> 2. proxychains4 wget 0.0.0.0:2345

i think this test is flawed too. you need to use e.g. proxychains4 wget 10.0.0.1:2345 where 10.0.0.1 is the local ip of your network adapter.

**Comment by NingYuanLin** [comment]
> > 2. proxychains4 wget 0.0.0.0:2345
> 
> i think this test is flawed too. you need to use e.g. proxychains4 wget 10.0.0.1:2345 where 10.0.0.1 is the local ip of your network adapter.

Yes.using your test method don't appear same problems. But why `proxychains4 wget 0.0.0.0:2345` also work in prior version?

**Comment by rofl0r** [comment]
> Yes.using your test method don't appear same problems

so, no problem at all? or if there's a problem is it what was described by OP? if so it would be helpful if you can bisect.

> But why proxychains4 wget 0.0.0.0:2345 also work in prior version?

by accident. 0.0.0.0 isn't a valid routable ip.

**Comment by NingYuanLin** [comment]
> > Yes.using your test method don't appear same problems
> 
> so, no problem at all? or if there's a problem is it what was described by OP? if so it would be helpful if you can bisect.

Yeah, there won't be any problems when using `proxychains4 wget 10.0.0.1:2345`

**Comment by NingYuanLin** [comment]
> by accident. 0.0.0.0 isn't a valid routable ip.

Well, maybe 0.0.0.0 isn't a valid routable ip. But pycharm's official pydevd tool is used in this way. I configured `localnet 0.0.0.0/0.0.0.0`. In version 4.15, a connection error occurred:
```
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/local//lib/libproxychains4.so
[proxychains] DLL init: proxychains-ng 4.15-git-1-g7de7dd0
Could not connect to 0.0.0.0: 44765
Traceback (most recent call last): 
.......
```
In version 4.12, it work well.
```
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/lib/x86_64-linux-gnu/libproxychains.so.4
[proxychains] DLL init: proxychains-ng 4.12
Connected to pydev debugger (build 212.5457.59)
```

**Comment by rofl0r** [comment]
> But pycharm's official pydevd tool is used in this way.

and where does the 0.0.0.0 come from ? is this something you configured, or how pycharm connects by default to its services ? the latter would be surprising.

**Comment by NingYuanLin** [comment]
> > But pycharm's official pydevd tool is used in this way.
> 
> and where does the 0.0.0.0 come from ? is this something you configured, or how pycharm connects by default to its services ? the latter would be surprising.

This is the default configuration of pycharm and I cannot easily change it unless I modify the source code. Thank you for your help, I don't want to continue tossing it now.
